source/Calamari.Azure/Integration/Websites/Publishing/ResourceManagerPublishProfileProvider.cs

```diff
@@ -43,12 +43,25 @@ public static SitePublishProfile GetPublishProperties(string subscriptionId, str
                     }
                 }
 
-                var matchingSite = sites
-                    .FirstOrDefault(webApp => string.Equals(webApp.Name, azureTargetSite.Site, StringComparison.CurrentCultureIgnoreCase) &&
-                                         (string.IsNullOrWhiteSpace(resourceGroupName) || string.Equals(webApp.ResourceGroup, resourceGroupName, StringComparison.InvariantCultureIgnoreCase)));
+                var matchingSites = sites
+                    .Where(webApp => string.Equals(webApp.Name, azureTargetSite.Site, StringComparison.CurrentCultureIgnoreCase) &&
+                                         (string.IsNullOrWhiteSpace(resourceGroupName) || string.Equals(webApp.ResourceGroup, resourceGroupName, StringComparison.InvariantCultureIgnoreCase)))
+                                         .ToList();
 
-                if (matchingSite == null)
+                if (!matchingSites.Any())
                     throw new CommandException($"Could not find Azure WebSite '{azureTargetSite.Site}' in subscription '{subscriptionId}'");
+                
+                // if more than one site, fail
+                if (matchingSites.Count > 1)
+                    throw new CommandException(
+                        $"Found {matchingSites.Count} matching the site name '{azureTargetSite.Site}' in subscription '{subscriptionId}'.{(string.IsNullOrWhiteSpace(resourceGroupName) ? " Please supply a Resource Group name." : string.Empty)}");
+
+                var matchingSite = matchingSites.First();
```

Comment:
I'd do `matchingSites.Single()` here, if the previous checks ever got incorrectly modified that would still catch the case where multiples are returned and prevent the deployment. Less friendly error, but at least doesn't incorrectly deploy to the wrong place.

Reply:
the less friendly message was what stopped me putting single here in the first place